### PR TITLE
feat: signer sponsorship

### PR DESCRIPTION
**Pull request by stephancill**
Adds farcaster signer sponsorship

### CODE REVIEW

File: apps/extension/hooks/use-farcaster-identity.tsx [hunk 1]
```diff
@@ -62,6 +62,10 @@ interface SignedKeyRequestBody {
   signature: string
   requestFid: string
   deadline: number
+  sponsorship: {
```
Comment (cryptods8) [inline apps/extension/hooks/use-farcaster-identity.tsx]: should this be optional?

File: apps/extension/hooks/use-farcaster-identity.tsx [hunk 2]
```diff
@@ -73,7 +77,7 @@ async function fetchCreateSignedKeyRequest(requestBody: SignedKeyRequestBody) {
   return result
 }
 
-interface SignerResponse {
+type SignerResponse = Omit<SignedKeyRequestBody, "key"> & {
```
Comment (cryptods8) [inline apps/extension/hooks/use-farcaster-identity.tsx:80]: can you explain the preference for `type`? (i am pretty much uninitiated in typescript and want to learn)

File: apps/extension/hooks/use-farcaster-identity.tsx (hunk 1 above)
Comment (stephancill) [reply apps/extension/hooks/use-farcaster-identity.tsx]: yes ty, fixed

File: apps/extension/hooks/use-farcaster-identity.tsx (hunk 2 above)
Comment (stephancill) [reply apps/extension/hooks/use-farcaster-identity.tsx:80]: the syntax is just easier when combining types, no real difference

**Comment by cryptods8** [comment]
also added the same for fc identity hook within server itself... ideally there shouldn't be this duplication, but i left it for later cleanup

File: apps/server/app/api/v1/signer/route.ts [hunk 3]
```diff
@@ -25,17 +31,18 @@ const SIGNED_KEY_REQUEST_TYPE = [
 
 const SIGNATURE_VALIDITY = 86400; // 1 day
 
-async function handleRequest(req: NextRequest) {
+export async function handler(req: NextRequest) {
```
Comment (cryptods8) [inline apps/server/app/api/v1/signer/route.ts]: this seems to be causing build issues

Comment (stephancill) [reply apps/server/app/api/v1/signer/route.ts]: oops my bad

**Pull request merged**